WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Bulk Edit Price/BulkUpdatePriceSettingsViewModel.swift

```diff
@@ -0,0 +1,144 @@
+    ///
+    enum EditingPriceType {
+        case regular
+        case sale
+
+        func keyPathForPriceType() -> KeyPath<ProductVariation, String?> {
+            switch self {
+            case .regular:
+                return \.regularPrice
+            case .sale:
+                return \.salePrice
+            }
+        }
+    }
+
+    /// The state of save price setting button
+    @Published private(set) var saveButtonState: ButtonState = .disabled
+
+    /// The error state
+    @Published private(set) var bulkUpdatePriceError: BulkUpdatePriceError? = nil
+
+    /// A Closure to be called when the price update is successful
+    private let priceUpdateDidFinish: () -> Void
+
+    private var currentPrice: String? = nil
+    private let siteID: Int64
+    private let productID: Int64
+    private let productVariations: [ProductVariation]
+    private let editingPriceType: EditingPriceType
+    private let storesManager: StoresManager
+    private let priceSettingsValidator: ProductPriceSettingsValidator
+
+    init(siteID: Int64,
+         productID: Int64,
+         productVariations: [ProductVariation],
+         editingPriceType: EditingPriceType,
+         priceUpdateDidFinish: @escaping () -> Void,
+         storesManager: StoresManager = ServiceLocator.stores,
+         currencySettings: CurrencySettings = ServiceLocator.currencySettings) {
+        self.siteID = siteID
+        self.productID = productID
+        self.productVariations = productVariations
+        self.priceUpdateDidFinish = priceUpdateDidFinish
+        self.editingPriceType = editingPriceType
+        self.storesManager = storesManager
+        self.priceSettingsValidator = ProductPriceSettingsValidator(currencySettings: currencySettings)
+    }
+
+    /// Called when the save button is tapped
+    ///
+    func saveButtonTapped() {
+        bulkUpdatePriceError = validatePrice()
+        guard bulkUpdatePriceError == nil else {
+            return
+        }
+
+        saveButtonState = .loading
+
+        let action = ProductVariationAction.updateProductVariations(siteID: siteID,
+                                                                    productID: productID,
+                                                                    productVariations: variationsWithUpdatedPrice()) { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .success(_):
+                self.priceUpdateDidFinish()
+            case let .failure(error):
+                DDLogError("⛔️ Error updating product variations: \(error)")
+                self.bulkUpdatePriceError = .priceUpdateError
+            }
+
+            self.saveButtonState = .enabled
+        }
+
+        storesManager.dispatch(action)
+    }
+
+    /// Called when price changes
+    ///
+    func handlePriceChange(_ price: String?) {
+        currentPrice = price
+        updateButtonStateBasedOnCurrentPrice()
+    }
```

Comment:
Is this to validate the price in real-time as the price is being input?

Reply:
Yes from mock ups I see the button has a disable state when we have no input. So this is to update the button state to enable/disable based on the current input.